File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/BooleanConverter.java
package org.matilda.commands.types;

import com.google.protobuf.BoolValue;

public class BooleanConverter implements ProtobufConverter<Boolean> {
    @Override
    public BoolValue convert(Boolean object) {
        return BoolValue.newBuilder().setValue(object).build();
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/BooleanConverter.kt
package org.matilda.commands.types

import pbandk.unpack
import pbandk.wkt.Any
import pbandk.wkt.BoolValue

class BooleanConverter : ProtobufConverter<Boolean> {
    override fun convertToProtobuf(obj: Boolean): BoolValue {
        return BoolValue(obj)
    }

    override fun convertFromProtobuf(obj: Any): Boolean {
        return obj.unpack(BoolValue.Companion).value
    }
}
